### PR TITLE
Add webhook support for Microsoft Teams

### PR DESCRIPTION
**Pull request by akshatdalton**
Fixes: #13395 and fixes: #12726.

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by chlins** [comment]
Thanks for your contribution, but let's hold on this PR as it conflict with webhook refactor in the v2.8.

**Review by chlins — CHANGES_REQUESTED** [review]
hold on

**Comment by chlins** [comment]
@akshatdalton Hi, the code base has been updated, could you rebase the main branch and rework your pr? Thanks!

**Comment by OrlinVasilev** [comment]
@akshatdalton hey mate can you check ^^ thanks!

**Comment by akshatdalton** [comment]
@chlins I've updated my PR please review!

**Comment by chlins** [comment]
@akshatdalton Thanks for your work, However, it seems that what you have modified and what we expect after refactoring are not quite consistent in terms of easily implementing a new payload format. In theory, you only need to implement a payload format interface instead of a new job. You can refer to the implementation of CloudEvents: https://github.com/goharbor/harbor/blob/main/src/pkg/notifier/formats/cloudevents.go

**Comment by wy65701436** [comment]
@akshatdalton Do you still have an interest in working on this PR? If so, we could consider marking it as a candidate for the v2.11.

**Comment by szpak** [comment]
> @akshatdalton Do you still have an interest in working on this PR? If so, we could consider marking it as a candidate for the v2.11.

@akshatdalton Will you time to finish this PR?

**Comment by szpak** [comment]
I would prefer to keep it open.

**Comment by szpak** [comment]
@akshatdalton Will you have time to finish this PR?

File: src/controller/webhook/controller_test.go [hunk 1]
```diff
@@ -99,6 +99,12 @@ func (c *controllerTestSuite) TestDeletePolicy() {
 	err = c.ctl.DeletePolicy(context.TODO(), 1)
```
Comment (Thesuperkingofsnakes22) [inline src/controller/webhook/controller_test.go:99]: **
```suggestion
 	err = c.ctl.DeletePolicy(context.TODO(), 1)@Vad1mo 
```
**

**Review by Thesuperkingofsnakes22 — APPROVED** [review]

**Pull request closed without merging**